### ISSUE TITLE
Grenade smoke still visible after ending test game

### ISSUE THREAD

**Issue by ZakJudges** [TheGameCreators/GameGuruRepo#1539]
![Screenshot 2021-11-12 092147](https://user-images.githubusercontent.com/32545391/141443212-1082df8a-9e58-4e7a-82fa-d9618c0e9300.png)

**Comment by ZakJudges** [comment]
@LeeBamberTGC  Would this one overlap with work that Maciej is doing at the moment? If so I can assign it to him :)

**Comment by LeeBamberTGC** [comment]
@ZakJudges Yes anything weapons-related can be assigned to @dllllllllllb (Maciej), and I will help things along as needed

**Comment by ZakJudges** [comment]
@dllllllllllb Steps to reproduce:
- throw a bunch of grenades in test level
- go back to editor
- there is usually one or two smoke effects still visible

**Comment by LeeBamberTGC** [comment]
@dllllllllllb I did not know off the top of my head, so I have documented the process I used to find the code. It is a good way to find stuff quickly in the vastness of the games engine:

1. First I look for a visual assets file, so in this case I know the grenade is a projectile, and the smoke is likely stored there, so I go here:
![image](https://user-images.githubusercontent.com/35840583/154066588-855811b5-f1a1-42c3-933b-84c62b63afc4.png)

2. Right enough, there is the smoke, so I look in the code for "projectiletypes"
3. Found the code which scans this folder to collect all projectiles, and each loop the function "weapon_projectile_load" is called
4. Looks like "t.WeaponProjectileBase[t.tNewProjBase].particleImageID" is used to store the loaded image
5. Looks like the only code that uses this value is in "weapon_add_projectile_particles"
6. This seems to call "ravey_particles_add_emitter" and so we can say that its this emitter which is creating the smoke
7. A quick scoot around the code and I find "ravey_particles_delete_all_emitters" function, but it is only called by the old MP code
8. I would say adding the "ravey_particles_delete_all_emitters()" call to the test level exit code will clean up all emitters
